src/daemon.py

```diff
@@ -0,0 +1,88 @@
+import os
+import time
+import pandas as pd
+from web3 import Web3
+from threading import Thread
+from sqlalchemy import create_engine
+from src.imbalances_script import RawTokenImbalances
+from src.config import CHAIN_RPC_ENDPOINTS, CHAIN_SLEEP_TIMES
+
+def get_web3_instance(chain_name):
+    return Web3(Web3.HTTPProvider(CHAIN_RPC_ENDPOINTS[chain_name]))
+
+def get_finalized_block_number(web3):
+    return web3.eth.block_number - 64
+
+def create_db_connection(chain_name):
+    """Helper function that creates a connection to the prod db."""
+    db_name = "mainnet" if chain_name == 'Ethereum' else "xdai"
+    db_url = os.getenv("PROD_DB_URL").format(chain=db_name)
+    
+    return create_engine(f"postgresql+psycopg2://{db_url}")
+
+def fetch_transaction_hashes(chain_name, start_block, end_block):
+    """Fetch transaction hashes beginning start_block."""
+    prod_connection = create_db_connection(chain_name)
+    query = f"""
+    SELECT tx_hash 
+    FROM settlements 
+    WHERE block_number >= {start_block} 
+    AND block_number<={end_block}
+    """
+
+    res = pd.read_sql(query, prod_connection)
+    
+    # converts hashes at memory location to hex 
+    res['tx_hash'] = res['tx_hash'].apply(lambda x: f"0x{x.hex()}")
+    
+    return res['tx_hash'].tolist()
+
+def process_transactions(chain_name):
+    web3 = get_web3_instance(chain_name)
+    rt = RawTokenImbalances(web3, chain_name)
+    sleep_time = CHAIN_SLEEP_TIMES.get(chain_name)
+
+    previous_block = get_finalized_block_number(web3)
+    unsuccessful_txs = []
+
+    print(f"{chain_name} Daemon started.")
+    
+    while True:
+        try:
+            latest_block = get_finalized_block_number(web3)
+            new_txs = fetch_transaction_hashes(chain_name, previous_block, latest_block)
+            all_txs = new_txs + unsuccessful_txs
+
+            for tx in all_txs:
+                print(f'Processing transaction on {chain_name}: {tx}')
+                try:
+                    imbalances = rt.compute_imbalances(tx)
+                    print(f"Token Imbalances on {chain_name}:")
```

Comment:
I guess we can take care of these in the follow-up PR that will fix the logs, but i would suggest we use a single multiline log msg in such cases so that no information is lost. The reason i am saying this is that sometimes, with multiple logs, the order gets messed up in kibana and then it's a bit of a pain to shuffle them around and map msgs to txs

Reply:
yes, I think I remember this happening during one of our first ebbo deployments. but yeah, can have this in the next pr.